cookieConsent/OneTrust/browser.js

```diff
@@ -0,0 +1,91 @@
+import logger from "../../utils/logUtil";
+
+/* eslint-disable class-methods-use-this */
+class OneTrust {
+  constructor(sourceConfig) {
+    this.sourceConfig = sourceConfig;
+  }
+
+  isEnabled(destConfig) {
+    // If user does not load onetrust sdk before loading rudderstack sdk
+    // we will not be filtering any of the destinations.
+    try {
+      if (!window.OneTrust || !window.OnetrustActiveGroups) {
+        return true;
+      }
+
+      /**
+     * Structure of onetrust consent group destination config.
+     * 
+     * "oneTrustConsentGroup": [
+                        {
+                            "oneTrustConsentGroup": "Performance Cookies"
+                        },
+                        {
+                            "oneTrustConsentGroup": "Functional Cookies"
+                        },
+                        {
+                            "oneTrustConsentGroup": ""
+                        }
+                    ]
+      *
+     */
+
+      const { oneTrustCookieCategories } = destConfig; // mapping of the destination with the consent group name
+
+      // If the destination do not have this mapping events will be sent.
+
+      if (!oneTrustCookieCategories) {
+        return true;
+      }
+      // OneTrust Cookie Compliance populates a data layer object OnetrustActiveGroups with
+      // the cookie categories that the user has consented to.
+      // Eg: ',C0001,C0003,'
+      // We split it and save it as an array.
+
+      const userSetConsentGroupIds = window.OnetrustActiveGroups.split(","); // Ids user has consented
+
+      // Get information about the cookie script - data includes, consent models, cookies in preference centre, etc.
+      // We get the groups(cookie categorization), user has created in one trust account.
+
+      const oneTrustAllGroupsInfo = window.OneTrust.GetDomainData().Groups;
+      const userSetConsentGroupNames = [];
+
+      // Get the names of the cookies consented by the user in the browser.
+
+      oneTrustAllGroupsInfo.forEach((group) => {
+        const { CustomGroupId, GroupName } = group;
+        if (userSetConsentGroupIds.includes(CustomGroupId)) {
+          userSetConsentGroupNames.push(GroupName.toUpperCase().trim());
+        }
+      });
```

Comment:
This logic can be moved to the constructor as it is not a destination dependant.

Reply:
done